File: Device/DeviceController.Data/Device.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DeviceController.Data
{
    public enum DeviceMode { Auto, Manual, Diagnostic }
    public enum DeviceState { Standby, Irrigating, Fault, Paused }
    public class Device
    {
        public int Id;
        public string Name;
        public string Description;
        public DeviceState State;
        public DeviceMode Mode;
        public string Status;
        public double Pressure;
        public double Flowrate;
        public int? ActiveProgramId;
        public int? IrrigationActionId;
        public IrrigationAction IrrigationAction;
        public string Inputs;
        public string Outputs;
        public string SoftwareVersion;        
        public string DeviceMAC;
        public int PumpSolenoidId;
        public DateTime CreatedAt;
        public DateTime UpdatedAt;       

        public Device()
        {           
        }
    }
}

File: WebClient/src/app/model/device.ts
import { IIrrigationAction } from "./irrigationaction";

export class IDevice {
    constructor(
        public id: number,
        public Name: string,
        public Description: string,
        public State: string,
        public Mode: string,
        public Status: string,
        public Pressure: Number,
        public PumpSolenoidId: number,
        public SoftwareVersion: string,
        public DeviceMAC: string,
        public ActiveProgramId: number,
        public IrrigationActionId: number,
        public IrrigationAction: IIrrigationAction,
        public createdAt: Date,
        public updatedAt: Date
    ) {}
}
